### ISSUE TITLE
How to train my own dataset not LRS2 ? - (for beginners) 

### ISSUE THREAD

**Issue by AlonDan** [Rudrabha/Wav2Lip#78]
First I must mention that I'm new to this and I'm not a programmer.
I did manage to use the pre-trained models checkpoints it wasn't so hard to follow even for a noob like me.

As the title I would like to train my own dataset and experiment with the procedure of what I should do.
I tried to follow the **Training on datasets other than LRS2** - section on the main page but it doesn't explain how to get started:
1. What files I need beisde my videos.mp4 files
2. Where to put the videos.mp4 files (special directory?)
3. What is the command I need to run, because I only saw LRS2 related:
```python color_syncnet_train.py --data_root lrs2_preprocessed/ --checkpoint_dir <folder_to_save_checkpoints>```
##
So I changed this **<folder_to_save_checkpoints>** to **checkpoints_NEW** which I put on the root directory of **Wav2Lip\checkpoints_NEW** 

But I'm confused about the other arguments, for example: ```--data_root lrs2_preprocessed/ ``` not sure what to replace this with since I don't want to use LRS2 but create my own dataset, that's my first confusion.

##
it says that training must be first for the '**expert discriminator**' I believe it's the: `lipsync_expert.pth` file which downloaded from the link and put inside the the directory 'checkpoints' with the 'wav2lip.pth' and 'wav2lip.pth'
so all the pre-trained models are in that directory, so far so good right?

So how do I even start now, I'm asking this because it is not very clear from the main page the more detailed steps are related to LRS2 which I'm not interested but nothing much about the `Training on datasets other than LRS2` 

## 
My Guessing experiments: (not sure if that's how it works)
I renamed my video.mp4 to 00001.mp4 and the next to higher numbers, I also created .txt files named:
**test**.txt, **train**.txt, **val**.txt  they are empty files but I think I need to put the video names or only their numbers? in some of them but I'm not sure exactly how it works (noob here) so I would like to get some step-by-step explanations from somebody who actually figure out how to train our own dataset from scratch, it will be very helpful to many others since unfortunately it's not easy to follow on the main page, especially for people who are new to this and would like to experiment.

##
Obviously a visual short video-tutorial or step-by-step example (from scratch) will be very helpful to follow and test but unfortunately I didn't find any tutorial out there to explain how to train our own datasets, that's why I had to ask here so hopefully it will help others with no experience who like to train their own datasets.

T**hanks ahead for ANY help** related, it will be very exciting to try train our own dataset.
I believe once the basics will be understood, it will only be a matter of detection (days of training which make sense and worth it!).

**Comment by Crazyjoedevola** [comment]
Hi. I had the same questions, so check the archived issues

**Comment by AlonDan** [comment]
> Hi. I had the same questions, so check the archived issues

Thanks! I did see your questions and answers thread but to be honest, it seems like you already figure out some basics that I didn't even know where to start with, so many things like I asked above such as:

- Creating dedicated specific directories?
- Creating empty .txt files, insert some name of mp4 files inside all of them? or only on part of them? - Do I need to download some specific files to put on specific directories?
- What commands do I need to use first? because there is some confusion about training 2 different models I think... 

And  So many other questions, but the most important thing is the correct order of step-by-step how to do it exactly, this is some mystery for newbies like myself.

I hope to see a tutorial in the future, unfortunately, I still didn't find anything like that related to training our own datasets.

**Comment by prajwalkr** [comment]
> but the most important thing is the correct order of step-by-step how to do it exactly

We suggest you try to reproduce everything using LRS2 first. this will help you get the basics right and you will be able to then try out other datasets.

**Comment by AlonDan** [comment]
> LRS2

I think I stuck when I tried to follow the instructions for LRS2 because I don't have access to the .txt files that needed to download via request which I don't think I can get because I'm not a student / big  research but only my very own private exploration to study and experiment.

Also, is LRS2 train on my own will be able to get high-res results or it's limited to what we have in the pre-trained models?

**Comment by Rudrabha** [comment]
I think you can get access to the LRS2 dataset as long as it is for research purposes. You can contact the mentioned email-id for the dataset. Also, training Wav2Lip on LRS2 is limited to the resolution that we already have in the pretrained models. If you want to train on your own data, please ensure it is saved in the exact folder structure like LRS2. Also ensure the videos are sampled at 25 fps. Once, you done this, the code should work without any hindrances.

**Comment by AlonDan** [comment]
This is the main reason why I would like to train my own dataset without LRS2, current pre-trained resolution is too low, also using my own videos (25 fps) still it is very hard for a newbie to follow the steps as I describe so I will have to wait for a more step-by-step tutorial or guide if anyone from the community will be ready to share their knowledge and experience with newbies like me and probably others who would like to learn how to train our own dataset, higher resolution, and most important how to actually do this from scratch.

I'm very curious to see actually high-res results in the future it will be amazing because Wav2Lip is already amazing! of course hopefully in the future it will support other than 90 degrees faces as well for much accurate results, but still it is an amazing project I hope to see some newbie-friendly tutorials for the training part not just for me but for newcomers who will be interested in this and probably will not be able to follow the steps on the main page as they're not for beginner's approach but for experienced people who understand most of the preparation and what to do.

Thanks ahead for anyone who helps in this :)

**Comment by Crazyjoedevola** [comment]
Yes, i would be very interested in a higher res version of the models as well

**Comment by rebotnix** [comment]
I also think that we have to use segmentation faces instead of rectangular faces to produce higher resolution output. I´m writing now the extract face part that makes generating of training content easier. I will share the source soon.  We can work all together on this to share this research.

**Comment by AlonDan** [comment]
@rebotnix t not that I understand the magic you'll do behind it but that sounds GREAT! I'll be happy to help testing / experiment once you'll make it happen, feel free to update your progress. ❤

But back to the main subject:
If anyone here have the experience in training our own data set (non LRS2)

Please consider to share a Step-by-Step (for noobs) tutorial or guide so we can follow it.
It will help to newcomers as well.
After so many guesses and reading the other posts, trying to bring the pieces together it's not simple for newbies with no experience but very simple for people who already know how it's done and did some tests.

So please, consider to make a guide / tutorial so we can all experiment and share our experience and help others while Wav2Lip will get better and better, so we can help testing it and maybe even share our own original datasets (banks) with the community as pre-trained models with higher resolution!

Thanks ahead and much love to every single one of you! 
keep up the good work and feel free to update this thread if you came up with a guide (or even a **Video Tutorial** on YouTube which could be MUCH easier to follow step-by-step)

**Comment by qustions** [comment]
> I also think that we have to use segmentation faces instead of rectangular faces to produce higher resolution output. I´m writing now the extract face part that makes generating of training content easier. I will share the source soon. We can work all together on this to share this research. I think we have to retrain SFD face detection as well for better accuracy from side faces.

@rebotnix you don't you try with retinaface it's more accurate and fast

**Comment by PMA65** [comment]
> I also think that we have to use segmentation faces instead of rectangular faces to produce higher resolution output. I´m writing now the extract face part that makes generating of training content easier. I will share the source soon. We can work all together on this to share this research. I think we have to retrain SFD face detection as well for better accuracy from side faces.

Do you need help on this ? 
I'm currently trying to implement this, any tips are appreciated 
Thank you

**Comment by nhutsam261** [comment]
> > I also think that we have to use segmentation faces instead of rectangular faces to produce higher resolution output. I´m writing now the extract face part that makes generating of training content easier. I will share the source soon. We can work all together on this to share this research. I think we have to retrain SFD face detection as well for better accuracy from side faces.
> 
> @rebotnix
> Do you need help on this ?
> I'm currently trying to implement this, any tips are appreciated
> Thank you

Have you done yet? can you share the result for training own dataset?
